panther_analysis_tool/cmd/configsdk_test.py

```diff
@@ -0,0 +1,280 @@
+        indirect_invocation: True if this function is being invoked as part of
+                                 another command (probably the legacy test command)
+
+    Returns:
+        A tuple of the return code, and a list of tuples containing invalid specs and their error.
+    """
+    panther_config_cache_path: Final = config_utils.get_config_cache_path()
+
+    try:
+        config_utils.run_config_module(panther_config_cache_path)
+        detection_intermediates: List[Dict] = config_utils.load_intermediate_config_cache(panther_config_cache_path)
+        detections: List[Detection] = [Detection(d) for d in detection_intermediates]
+        detections = _filter_detections(args, detections)
+        logging.info('Running Unit Tests for Panther Content\n')
+        tests_failed = _run_unit_tests(detections, args.minimum_tests)
+        return int(not tests_failed), []
+    except FileNotFoundError as err:
+        if indirect_invocation:
+            # If this is run automatically at the end of the standard test command,
+            # this isn't an error that should cause the invocation to return 1.
+            logging.debug(err)
+            return 0, []
+        logging.error(err)
+        return 1, []
+
+
+def _filter_detections(args: argparse.Namespace, detections: List[Detection]) -> List[Detection]:
+    """Filters out the detections to be tested by using the command line args.
+
+    Args:
+        args: The populated Argparse namespace with parsed command-line arguments.
+        detections: The list of detections that need to be filtered for testing.
+
+    Returns:
+        A list of detections to be unit tested.
+    """
+    filtered = []
+    for detection in detections:
+        # filter out detections with no unit tests
+        if not detection.has_unit_tests():
+            continue
+
+        # filter out using filter arg TODO
+
+        # filter using enabled only arg
+        if args.skip_disabled_tests and detection.disabled():
+            continue
+
+        filtered.append(detection)
+
+    return filtered
+
+
+def _run_unit_tests(detections: List[Detection], min_tests: int = 0) -> bool:
+    """Runs the unit tests for the given detections, printing out test results and a summary.
+
+    Args:
+        detections: The list of detections to be unit tested.
+        min_tests: The minimum number of unit tests each detection needs to have.
+
+    Returns:
+        True if at least one unit test failed, False if all tests pass.
+    """
+    for detection in detections:
+        print(detection.detection_id)
+        if len(detection.filters) == 0:
+            print('    Detection had no filters to test')
+            continue
+
+        for unit_test in detection.unit_tests:
+            prg = unit_test.get_prg(detection.filters, detection.detection_type)
+            locs: Dict[str, str] = {}
+            exec(prg, {}, locs)  # nosec B102 pylint: disable=W0122
```

Comment:
I think i'm missing something here, how does the mock event get passed in?  I'm not familiar with config sdk unit tests

Reply:
The event is written as a variable as a part of the program. The evaluate function gets called using the variable. The result is saved in the `locs` dictionary